### PR TITLE
offlineimap: fix audit errors and plist issues

### PR DESCRIPTION
**Pull request by hvoigt**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

This is based on a seperate pull request fixing the existing audit errors: https://github.com/Homebrew/homebrew-core/pull/1008 which needs to be reviewed and merged first.

### CODE REVIEW

File: Formula/offlineimap.rb [hunk 1]
```diff
@@ -24,15 +24,18 @@ def caveats; <<-EOS.undent
...
 
   def plist; <<-EOS.undent
     <?xml version="1.0" encoding="UTF-8"?>
     <!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
     <plist version="1.0">
       <dict>
+        <key>EnvironmentVariables</key>
+        <dict>
+          <key>PATH</key>
+          <string>/usr/bin:/usr/local/bin</string>
```
Comment (mikemcquaid) [inline Formula/offlineimap.rb]: Are these not there by default?

File: Formula/offlineimap.rb [hunk 2]
```diff
@@ -24,15 +24,18 @@ def caveats; <<-EOS.undent
     EOS
   end
 
-  test do
-    system bin/"offlineimap", "--version"
-  end
+  plist_options :startup => true
```
Comment (mikemcquaid) [inline Formula/offlineimap.rb]: Needs a `:manual` bit and I'm not sure this needs to be `:startup`?

File: Formula/offlineimap.rb (hunk 1 above)
Comment (hvoigt) [inline Formula/offlineimap.rb]: No, /usr/local/bin is not there by default.

File: Formula/offlineimap.rb (hunk 2 above)
Comment (hvoigt) [inline Formula/offlineimap.rb]: What does manual actually mean? I figured startup means that the service should be started upon login if configured by the user and manual describes some manual method the user can use to manually start the service. What command should go into the manual part just the same as the one in the plist without the full path?

File: Formula/offlineimap.rb (hunk 1 above)
Comment (mikemcquaid) [inline Formula/offlineimap.rb]: Ok. You probably want `#{HOMEBREW_PREFIX}/bin` or `#{HOMEBREW_PREFIX}/opt/offlineimap/bin`. Was it broken before?

Comment (hvoigt) [inline Formula/offlineimap.rb]: It was not broken, but as explained in the commit message it was inconvenient that you could not call homebrew tools from offlineimap without full path. E.g. notmuch which indexes all messages for quicker searching.

File: Formula/offlineimap.rb (hunk 2 above)
Comment (hvoigt) [inline Formula/offlineimap.rb]: The only documentation I found for plist_options is [this](http://www.rubydoc.info/github/Homebrew/brew/Formula.plist_options). But it does not really explain this option for me.

File: Formula/offlineimap.rb (hunk 1 above)
Comment (mikemcquaid) [inline Formula/offlineimap.rb]: Ok, that's cool, thanks.

File: Formula/offlineimap.rb (hunk 2 above)
Comment (mikemcquaid) [inline Formula/offlineimap.rb]: Yeh, that's the best docs we have for now (but we should improve them). In this case it doesn't need `:startup` and should specify a `:manual` on how to run this outside `launchd`

**Comment by hvoigt** [comment]
Ok pushed another version with the requested changes incorporated. I additionally checked what the default path set by launchd is (/usr/bin:/bin:/usr/sbin:/sbin) and added those as well so we do not break any existing use cases of others.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request merged**